server/utils.py

```diff
@@ -110,6 +103,9 @@ def _generate_processed_compose_file(
             "services"
         ]["nginx"]
 
+        self._write_compose_file()
+
+    def _write_compose_file(self):
         with open(self._compose_file_location, "w") as yaml_file:
```

Comment:
This can cause an `exception`, right? Shouldn't this be in `try/catch`?

Reply:
Yes, a lot of error handling is missing. I intended for the action to fail - but I should probably catch these errors and send a proper status code with an error message rather than GitHub Action receiving a 500 error 🤔

Reply:
Yep, some kind of handling will be nice! I would suggest having a custom exception so your logs are cleaner, like: `FileOpenException`.

Reply:
Yes, I should have custom exceptions. Thanks, I will do the whole exception refactoring in a separate PR.